static/app/utils/profiling/renderers/shaders.tsx

```diff
@@ -0,0 +1,67 @@
+import {FlamegraphTheme} from '../flamegraph/FlamegraphTheme';
+
+export const vertex = () => `
+attribute vec2 a_position;
+attribute vec4 a_color;
+attribute vec4 a_bounds;
+
+uniform mat3 u_model;
+uniform mat3 u_projection;
+
+varying lowp vec4 v_color;
+varying vec2 v_pos;
+varying vec4 v_bounds;
+
+void main() {
+  vec2 scaled = (u_model * vec3(a_position.xy, 1)).xy;
+  vec2 pos = (u_projection * vec3(scaled.xy, 1)).xy;
+
+  gl_Position = vec4(pos, 0.0, 1.0);
+
+  v_color = a_color;
+  v_pos = a_position.xy;
+  v_bounds = a_bounds;
+}
+`;
+
+export const fragment = (theme: FlamegraphTheme) => `
+// fragment shaders don't have a default precision so we need
+// to pick one. mediump is a good default
+precision mediump float;
+
+uniform bool u_is_search_result;
+uniform vec2 u_border_width;
+uniform bool u_draw_border;
+
+varying lowp vec4 v_color;
+varying vec4 v_bounds;
+varying vec2 v_pos;
+
+void main() {
+  float minX = v_bounds.x + u_border_width.x;
+  float maxX = v_bounds.z - u_border_width.x - v_bounds.x;
```

Comment:
I noticed that `v_bounds.z` is initialized as

```typescript
this.bounds[offset + 2] = frame.start + frame.end;
```

while `v_bounds.x` is initialized as

```typescript
this.bounds[offset] = x1 = frame.start;
```

If I understood this correctly, you should be able to initialize `v_bounds.z` as

```typescript
this.bounds[offset + 2] = x2;
```

and remove the extra subtraction here to achieve the same results.

```suggestion
  float maxX = v_bounds.z - u_border_width.x;
```

I'm still reading through WebGL so let me know if I misunderstood anything here.